app/Http/Controllers/Api/V1/CustodianHasProjectUserController.php

```diff
@@ -372,4 +381,36 @@ public function getWorkflowTransitions(Request $request)
     {
         return $this->OKResponse(CustodianHasProjectUser::getTransitions());
     }
+
+    public function sendNotifications(int $custodianId, int $projectUserId, ?string $newState, ?string $oldState)
+    {
+        $projectUser = ProjectHasUser::where('id', $projectUserId)->first();
+        $project = Project::where('id', $projectUser->project_id)->first();
+        $registry = Registry::where('digi_ident', $projectUser->user_digital_ident)->first();
+        $user = User::where('registry_id', $registry->id)->first();
+        $affiliation = Affiliation::where('registry_id', $registry->id)->first();
+        $organisation = Organisation::where('id', $affiliation->organisation_id)->first();
+        $userOrganisation = User::where('organisation_id', $affiliation->organisation_id)->first();
+        $userCustodian = User::where('custodian_user_id', $custodianId)->first();
```

Comment:
Sorry Dan, I'm gonna go down a similar route. I can see _why_ you're doing it this way, but you can do this in a single query, instead of 8. This is massively expensive call.

You can either use raw sql, with left/inner joins to dictate the state of the data you need for each user type, or you can use relations, which are automatically nulled if not applicable.

Don't get me wrong. I'm all for hundreds of really quick db calls, instead of 1 massive one, but these can be handled a little better.